graph_snyk_cve_sync.py

```diff
@@ -100,55 +103,86 @@ def force_run_ingestion(self):
         """Return if ingestion mode is on."""
         return os.environ.get('SNYK_INGESTION_FORCE_RUN', 'false').lower() in ('1', 'yes', 'true')
 
+    def is_delta_mode_on(self):
+        """Return if the delta feed mode is on."""
+        return os.environ.get('SNYK_DELTA_FEED_MODE', 'false').lower() in ('1', 'yes', 'true')
+
 
 class SnykCveSync:
     """Snyk class to sync cves to graph."""
 
-    def __init__(self):
+    def __init__(self, data):
         """Init method for helper class."""
         self.helper = Helper()
         self.CVE_DATA = {}
-
-    def get_version_rules_npm(self, vuln_versions):
-        """Version rules for npm eco."""
-        rules = []
-        for version in vuln_versions:
-            rules.append(NpmSpec(version))
-        return rules
+        self.utc_now = datetime.utcnow()
+        if not data:
+            self.snyk_data = self.helper.read_data_from_s3(self.utc_now.strftime('%d-%m-%y'),
+                                                           "snyk-feed/")
+        else:
+            self.snyk_data = data
+
+    def parse_data_for_eco(self, eco):
+        """Return True/False depending on whether the ecosysytem data should be parsed or not."""
+        if self.helper.is_delta_mode_on():
+            return True
+        day = str(self.utc_now.weekday())
+        if ((day == "0" or day == "3") and eco == "java") \
+                or ((day == "1" or day == "4") and eco == "js") \
+                or ((day == "2" or day == "5") and eco == "python"):
+            return True
+        return False
 
     def get_version_rules(self, vuln_versions):
```

Comment:
Can all the private methods start with a _ to follow python conventions.

Reply:
problem with making it pvt is that we cannot use it in the unit test cases. If we want to test very minute functionalities limited to a particular function, we can add a test around that. If we make it pvt, then we need to run the entire flow in order to test and its time consuming as well as it tends to miss out on few conditions. Hence we follow the idea of keeping it public and then making use of extensive testing against the same. You can check that in the unit test files if needed

Reply:
I dont think this approach is correct. Private methods will be tested through the public methods from where they are called. Also in the case of Python, it is just a naming convention and there is not restriction on calling it. 
I think it is good to follow the Python conventions.

Reply:
no. as soon as you make it private, we cannot call it exclusively from unit test. Actually there is a history to it why we have moved to this approach.